### ISSUE TITLE
Cannot filter on child model property without enabling property in parent [Filter]

### ISSUE THREAD

**Issue by robinkanters** [OData/WebApi#2261]
I'm trying to filter on a child collection of my OData model by adding `[Filter]` attributes. I have added the `[Filter]` attribute to my top-level model, and that works. However I am getting an error message saying that the property on my child model cannot be used as a filter unless I add it to the attribute on my main model.

### Assemblies affected

*Which assemblies and versions are known to be affected e.g. OData .Net lib 7.x*
- System.Web.OData v6.0.0.0 _(I think)_

### Reproduce steps

Filter string: `children/any(x: x/name eq 'bar')`

Controller and models:

```csharp
[ODataRoutePrefix("repro")]
public class ReproController : ODataController
{
    public static List<ModelA> Data { get; set; } = new List<ModelA>
    {
        new ModelA(),
    };

    [HttpGet]
    [ODataRoute]
    [EnableQuery(AllowedQueryOptions = AllowedQueryOptions.Filter)]
    public Task Repro(ODataQueryOptions<ModelA> options)
    {
        options.Validate(new ODataValidationSettings
        {
            AllowedQueryOptions = AllowedQueryOptions.Filter,
        });

        return Task.CompletedTask;
    }
}

[Filter("children")]
public class ModelA
{
    [Key]
    public string Id { get; set; } = Guid.NewGuid().ToString();

    public IEnumerable<ModelB> Children { get; set; } = new List<ModelB>
    {
        new ModelB(),
    };
}

[Filter("name")]
public class ModelB
{
    public string Name { get; set; } = "foo";
}
```

### Expected result

- Filter validation succeeds.

### Actual result

- Validation fails: `The property 'name' cannot be used in the $filter query option.`

### Additional detail

It does work when I add `"name"` to the `Filter` attribute on `ModelA`, but that doesn't seem like it should be necessary.

**Comment by mikepizzo** [comment]
@robinkanters  - right you are.  transferring the issue.  I note that there is already an issue in that repo, so we may eventual (re-)close this as a dupe, but wanted to keep this one open for now for the active discussion.

**Comment by mikepizzo** [comment]
Related issue #1696 has a detailed investigation of this issue. Perhaps we can get it fixed for 7.7.1.
